lib/ex_doc/formatter/html/templates.ex

```diff
@@ -209,7 +211,7 @@ defmodule ExDoc.Formatter.HTML.Templates do
   @spec header_to_id(String.t) :: String.t
   def header_to_id(header) do
     header
-    |> String.replace(~r/<.+>/, "")
+    |> String.replace(@clean_html_regex, "")
```

Comment:
This function is used separately in templates to generate ID.
Therefore proper clean-up is needed here too